homeassistant/components/flipr/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow for Flipr integration."""
+from typing import List
+
+from flipr_api import FliprAPIRestClient
+from requests.exceptions import HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import CONF_FLIPR_ID, CONF_PASSWORD, CONF_USERNAME
+from .const import DOMAIN  # pylint:disable=unused-import
+from .crypt_util import encrypt_data
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Flipr."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Flipr config flow."""
+        self._username = None
+        self._password = None
+        self._flipr_id = None
+        self._possible_flipr_ids = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self._show_setup_form()
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+
+        if not self._flipr_id:
+            try:
+                flipr_ids = await self._authenticate_and_search_flipr()
+            except HTTPError:
+                errors = {}
+                errors["base"] = "invalid_auth"
+                return self._show_setup_form(errors)
```

Comment:
Actually, I do not know how the server behave in bad situations since I have only see it up and running.
I added your proposal in the code to catch other errors to improve it.